### PR TITLE
Add wlr_output::enabled

### PR DESCRIPTION
**Pull request by emersion**
This un-exports `wlr_output_{create,destroy}_global`, since this is a protocol matter. This instead adds a `wlr_output_update_enabled` for backends.
  
This fixes disabled outputs with the DRM backend.

- [x] Fix cursor at (0, 0) when re-enabling an output. This happens because the output layout moves the cursor to (0, 0) when the output is disabled.
- [x] Fix failed pageflip. This happens when re-enabling the output, no idea how to fix it.

Test plan:

```ini
[bindings]
Logo+t = toggle_outputs
```

See @acrisci's comment for edge cases.

Related Weston patch: https://patchwork.freedesktop.org/patch/194357/

### CODE REVIEW

File: include/wlr/interfaces/wlr_output.h [hunk 1]
```diff
@@ -26,14 +26,11 @@ struct wlr_output_impl {
 };
 
 void wlr_output_init(struct wlr_output *output, struct wlr_backend *backend,
-	const struct wlr_output_impl *impl);
-void wlr_output_free(struct wlr_output *output);
```
Comment (martinetd) [inline include/wlr/interfaces/wlr_output.h]: No wlr_output_destroy or similar to replace that?

File: include/wlr/interfaces/wlr_output.h [hunk 2]
```diff
@@ -26,14 +26,11 @@ struct wlr_output_impl {
...
+	const struct wlr_output_impl *impl, struct wl_display *display);
 void wlr_output_update_mode(struct wlr_output *output,
 	struct wlr_output_mode *mode);
 void wlr_output_update_custom_mode(struct wlr_output *output, int32_t width,
 	int32_t height, int32_t refresh);
-struct wl_global *wlr_output_create_global(struct wlr_output *wlr_output,
-	struct wl_display *display);
-void wlr_output_destroy_global(struct wlr_output *wlr_output);
+void wlr_output_update_enabled(struct wlr_output *output, bool enabled);
```
Comment (martinetd) [inline include/wlr/interfaces/wlr_output.h]: I'm saying that without thinking, would it make sense for someone to disable the output, and re-enable it later with a different display than the one it was init with? I guess it's better for them to create a different output in that case, but I'm not really sure what display represents so just asking.

File: include/wlr/interfaces/wlr_output.h (hunk 1 above)
Comment (emersion) [reply include/wlr/interfaces/wlr_output.h]: There's `wlr_output_destroy` in `include/wlr/types/wlr_output.h`. `wlr_output_free` was not implemented.

**Comment by acrisci** [comment]
Can you add some configuration for this in rootston to make it easier to demonstrate the feature?

There should be a config option to start with the output disabled, and maybe a command to enable/disable outputs at runtime.

**Comment by emersion** [comment]
Disabling outputs with the DRM backend fails with `Atomic test failed: Invalid argument`. I've try to debug that, but I don't understand why it fails (cc @ascent12).

**Comment by emersion** [comment]
Also, how should the output layout react to disabled outputs? Allocate a 0x0 area?

**Comment by acrisci** [comment]
Edge cases I can think of without testing:

* We need to automatically update the output layout when the output is enabled/disabled (make sure the cursor does not go on the disabled output).
* Make sure the output frame callback doesn't do anything
* Move all the windows from the disabled output to an enabled one in rootston
* Make sure the enabled state persists when switching VTs

**Comment by acrisci** [comment]
> Also, how should the output layout react to disabled outputs?

I think we should just treat it like it's not in the layout at all when it concerns any coordinate queries. The auto layout should reflow to reflect the output not being there. But leave it in there and make it come back when it gets enabled again.

**Comment by emersion** [comment]
I think this is ready for a first review. Testers: there are two known issues detailed in the first comment.

**Comment by emersion** [comment]
Concerning the output layout, as discussed on IRC, it makes more sense not to do anything special for disabled outputs. A compositor might want to remove an output from the output layout when disabled, but if a screen is turned off because the user is idle the compositor might not want the windows and the cursor to move.

**Comment by emersion** [comment]
Bump

**Comment by acrisci** [comment]
Did you fix the failed page flip issue?

**Comment by emersion** [comment]
>Did you fix the failed page flip issue?

Fixed!
